File: sahanadara-api/src/main/java/lk/chathurabuddi/sahanadara/api/entity/ReliefRequest.java
package lk.chathurabuddi.sahanadara.api.entity;

import javax.persistence.Entity;
import javax.persistence.ManyToOne;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Data
@EqualsAndHashCode(callSuper = true)
public class ReliefRequest extends BaseEntity {

    private String title;

    private String description;

    @ManyToOne
    private User requestingUser;

    @ManyToOne
    private Incident associatedIncident;

    @ManyToOne
    private Location location;

    private boolean completed;
}

File: sahanadara-ui/src/app/relief-request.ts
import { Location } from './location';
import { AssociatedIncident } from './associated-incident';
import { RequestingUser } from './requesting-user';

export interface ReliefRequest {
    id: number;
    creationTimestamp: Date;
    updateTimestamp: Date;
    title: string;
    description: string;
    requestingUser: RequestingUser;
    associatedIncident: AssociatedIncident;
    location: Location;
    completed: boolean;
}
